plugin/plugin.go

```diff
@@ -164,3 +173,43 @@ func (c execCommander) Output(ctx context.Context, name string, command proto.Co
 	}
 	return stdout.Bytes(), nil, nil
 }
+
+// validate checks if the metadata is correctly populated.
+func validate(metadata *proto.GetMetadataResponse) error {
+	if metadata.Name == "" {
+		return errors.New("empty name")
+	}
+	if metadata.Description == "" {
+		return errors.New("empty description")
+	}
+	if metadata.Version == "" {
+		return errors.New("empty version")
+	}
+	if metadata.URL == "" {
+		return errors.New("empty url")
+	}
+	if len(metadata.Capabilities) == 0 {
+		return errors.New("empty capabilities")
+	}
+	if len(metadata.SupportedContractVersions) == 0 {
+		return errors.New("empty supported contract versions")
+	}
+	if !supportsContract(proto.ContractVersion, metadata) {
+		return fmt.Errorf(
+			"contract version %q is not in the list of the plugin supported versions %v",
+			proto.ContractVersion, metadata.SupportedContractVersions,
+		)
+	}
+	return nil
+}
+
+// supportsContract return true if the metadata states that the
+// contract version is supported.
+func supportsContract(targetVer string, metadata *proto.GetMetadataResponse) bool {
+	for _, v := range metadata.SupportedContractVersions {
+		if v == targetVer {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
We can leverage `internal/slices.Contains` instead.

Reply:
Updated.